includes/classes/Indexable/User/User.php

```diff
@@ -173,6 +173,22 @@ public function format_args( $query_vars, $query ) {
 
 					$use_filters = true;
 				}
+
+				// If there are no specific roles named, make sure the user is a member of the site.
+				if ( ! $use_filters ) {
```

Comment:
After I submitted my patch (see #2601), I was wondering if relying on `$use_filters` was a good idea. Right now, this works fine, just because there are no conditions that would set `$use_filters` to `true` except for the ones related to roles. However, if there ever was some new code to be added after the inital `$use_filters = false;` and the roles conditions, this would no longer be true.

Instead of checking `$use_filters`, one could do it like so (which is a little more verbose, but absolutely specific to what we're after):

```suggestion
				if ( empty( $query_vars['role'] ) && empty( $query_vars['role__in'] ) && empty( $query_vars['role__not_in'] ) ) {
```

This should work just fine with your tests as well.

Reply:
You have a good point @tfrommen. I went ahead and used that to do a small refactor in the entire if block. Hopefully, it'll make the code more future-proof and also easier to be understood by new programmers. Thanks!